lib/tasks/buildThemes.js

```diff
@@ -128,19 +166,45 @@ export default async function({
 				log.verbose(`Available sap.ui.core themes: ${availableThemes.join(", ")}`);
 			}
 		}
-		allResources = allResources.filter(isAvailable);
+		themeResources = themeResources.filter(isAvailable);
+	}
+
+	let buildTheme;
+	if (useWorkers) {
+		if (!taskUtil) {
+			// TaskUtil is required for worker support
+			throw new Error(`buildThemes: Option 'useWorkers' requires a taskUtil instance to be provided`);
+		}
+		buildTheme = buildThemeInWorker;
+	} else {
+		// Do not use workerpool
+		buildTheme = (await import("../processors/themeBuilderWorker.js")).default;
 	}
 
-	const processedResources = await themeBuilder({
-		resources: allResources,
-		fs: fsInterface(combo),
+	const allResources = await combo.byGlob("/resources/**/*.less");
+	const makeTransferableResource = async (resourceCollection) =>
+		Promise.all(
+			resourceCollection.map(async (res) => {
+				return {
+					string: await res.getString(),
+					name: res.getName(),
+					path: res.getPath(),
+				};
+			})
+		);
+
+	const processedResources = await buildTheme({
+		allResources: await makeTransferableResource(allResources),
+		themeResources: await makeTransferableResource(themeResources),
```

Comment:
I expect it to be very expensive to always transfer all files to the worker. What about just requesting the needed files from within the worker instead? I'd like to see how that would compare performance-wise.